src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,440 @@
+
+    using EnumerableSet for EnumerableSet.UintSet;
+    using SafeMath for uint256;
+
+    ///////////////////////////////  Data //////////////////////////////
+
+    struct Data {
+        EnumerableSet.UintSet _assets;
+        mapping(uint256 => uint256) _values;
+    }
+
+    uint256 public _nextId;
+
+    address internal _minter;
+    AssetToken _asset;
+
+    bytes4 private constant ERC1155_RECEIVED = 0xf23a6e61;
+    bytes4 private constant ERC1155_BATCH_RECEIVED = 0xbc197c81;
+    uint256 private constant CREATOR_OFFSET_MULTIPLIER = uint256(2)**(256 - 160);
+
+    mapping(uint256 => Data) private _gameData;
+    mapping(address => address) private _creatorship; // creatorship transfer
+
+    mapping(uint256 => string) private _metaData;
+    mapping(uint256 => mapping(address => bool)) private _gameEditors;
+
+    ///////////////////////////////  Events //////////////////////////////
+
+    event Minter(address newMinter);
+    event AssetsAdded(uint256 indexed id, uint256[] assets, uint256[] values);
+    event AssetsRemoved(uint256 indexed id, uint256[] assets, uint256[] values, address to);
+    event CreatorshipTransfer(address indexed original, address indexed from, address indexed to);
+
+    constructor(
+        address metaTransactionContract,
+        address admin,
+        AssetToken asset
+    ) ERC721BaseToken(metaTransactionContract, admin) {
+        _asset = asset;
+        _nextId = 1;
+    }
+
+    ///////////////////////////////  Modifiers //////////////////////////////
+
+    modifier minterGuard() {
+        require(msg.sender == _minter || _minter == address(0), "INVALID_MINTER");
+        _;
+    }
+
+    modifier onlyOwnerOrEditor(uint256 id) {
+        require(msg.sender == _ownerOf(id) || _gameEditors[id][msg.sender], "OWNER_EDITOR_ACCESS_DENIED");
+        _;
+    }
+
+    modifier notToZero(address to) {
+        require(to != address(0), "DESTINATION_ZERO_ADDRESS");
+        _;
+    }
+
+    ///////////////////////////////  Functions //////////////////////////////
+
+    /// @notice Function to remove multiple assets from a GAME
+    /// @param gameId The GAME to remove assets from
+    /// @param assetId The asset Id to remove
+    /// @param to The address to send the removed asset to
+    /// @param uri The URI string to update the GAME token's URI
+    function removeSingleAsset(
+        uint256 gameId,
+        uint256 assetId,
+        address to,
+        string memory uri
+    ) external override minterGuard() onlyOwnerOrEditor(gameId) notToZero(to) {
+        _gameData[gameId]._values[assetId] = _gameData[gameId]._values[assetId].sub(1);
+        uint256 remainingAssets = _gameData[gameId]._values[assetId];
+
+        if (remainingAssets == 0) {
+            // "remove" is from EnumerableSet.sol
+            require(_gameData[gameId]._assets.remove(assetId), "ASSET_NOT_IN_GAME");
+        }
+
+        _asset.safeTransferFrom(address(this), to, assetId);
```

Comment:
it should use ERC1155 safeTransferFrom else it will only work for ERC721

Reply:
Thanks, good catch. Fixed.